keras_contrib/layers/capsule.py

```diff
@@ -0,0 +1,177 @@
+
+           conv_2d = Conv2D(64,
+                            (3, 3),
+                            activation='relu')(input_image)
+
+           capsule = Capsule(num_capsule=10,
+                             dim_capsule=16,
+                             routings=3,
+                             share_weights=True)(conv_2d)
+
+           2). NLP
+
+           maxlen = 72
+           max_features = 120000
+           input_text = Input(shape=(maxlen,))
+
+           embedding = Embedding(max_features,
+                                 embed_size,
+                                 weights=[embedding_matrix],
+                                 trainable=False)(input_text)
+
+           bi_gru = Bidirectional(GRU(64,
+                                      return_seqeunces=True))(embedding)
+
+           capsule = Capsule(num_capsule=5,
+                             dim_capsule=5,
+                             routings=4,
+                             share_weights=True)(bi_gru)
+
+       # Arguments
+           num_capsule : Number of Capsules (int)
+           dim_capsules : Dimensions of the vector output of each Capsule (int)
+           routings : Number of dynamic routings in the Capsule Layer (int)
+           share_weights : Whether to share weights between Capsules or not
+           (boolean)
+
+       # Input shape
+            3D tensor with shape:
+            (batch_size, input_num_capsule, input_dim_capsule)
+            [any 3-D Tensor with the first dimension as batch_size]
+
+       # Output shape
+            3D tensor with shape:
+            (batch_size, num_capsule, dim_capsule)
+
+       # References
+        - [Dynamic-Routing-Between-Capsules]
+          (https://arxiv.org/pdf/1710.09829.pdf)
+        - [Keras-Examples-CIFAR10-CNN-Capsule]"""
+
+    def __init__(self,
+                 num_capsule,
+                 dim_capsule,
+                 routings=3,
+                 share_weights=True,
+                 activation='squash',
+                 **kwargs):
+        super(Capsule, self).__init__(**kwargs)
+        self.num_capsule = num_capsule
+        self.dim_capsule = dim_capsule
+        self.routings = routings
+        self.share_weights = share_weights
+        if activation == 'squash':
+            self.activation = squash
+        else:
+            self.activation = activations.get(activation)
+
+    def build(self, input_shape):
+        super(Capsule, self).build(input_shape)
+        input_dim_capsule = input_shape[-1]
+        if self.share_weights:
+            self.W = self.add_weight(name='capsule_kernel',
+                                     shape=(1,
+                                            input_dim_capsule,
+                                            self.num_capsule *
+                                            self.dim_capsule),
+                                     initializer='glorot_uniform',
+                                     trainable=True)
```

Comment:
No need to specify the default value here.

Reply:
Changed it to 'None'.